__init__.py

```diff
@@ -10,158 +10,116 @@
+                                 # if a gui is available the title is read and image displayed
+
+    # explicit intents
     @intent_handler("search_wolfie.intent")
     def handle_search(self, message):
         query = message.data["query"]
         response = self.ask_the_wolf(query)
         if response:
-            self.speak(response)
+            self.speak_result()
         else:
             self.speak_dialog("no_answer")
 
+    @intent_handler(IntentBuilder("WolfieMore").require("More").
+                    require("WolfieKnows"))
+    def handle_tell_more(self, message):
+        """ Follow up query handler, "tell me more"."""
+        self.speak_result()
+
+    # common query integration
     def CQS_match_query_phrase(self, utt):
         self.log.debug("WolframAlpha query: " + utt)
         response = self.ask_the_wolf(utt)
         if response:
+            self.idx += 1  # spoken by common query framework
             return (utt, CQSMatchLevel.GENERAL, response,
                     {'query': utt, 'answer': response})
 
-    def ask_the_wolf(self, query):
-        # Automatic translation to English
-        utt = self.translate(query, "en", self.lang)
-
-        if self.settings["use_selene"]:
-            response = self.get_selene_response(utt)
-        elif self.settings.get("api_key"):
-            response = self.get_wolfram_response(utt)
-        else:
-            return None
+    def CQS_action(self, phrase, data):
+        """ If selected show gui """
+        self.display_wolfie()
 
-        if not response:
-            return None
-
-        bad_answers = ["No spoken result available",
-                       "Wolfram Alpha did not understand your input"]
-        if response in bad_answers:
-            return None
-
-        return self.translate(response)
-
-    def get_selene_response(self, query):
-        if query in self.answer_cache:
-            answer = self.answer_cache[query]
-        else:
-            answer = self.selene_api.spoken(
-                query=query,
-                lat_lon=(self.location['coordinate']['latitude'],
-                         self.location['coordinate']['longitude']),
-                units=self.config_core['system_unit'])
-            self.answer_cache[query] = answer
-        return answer
-
-    def get_wolfram_response(self, query):
-        url = 'http://api.wolframalpha.com/v1/spoken'
-        if query in self.answer_cache:
-            answer = self.answer_cache[query]
+    # wolfram integration
+    def ask_the_wolf(self, query):
+        # context for follow up questions
+        self.set_context("WolfieKnows", query)
+        self.results = self.wolfie.long_answer(query)
+        self.idx = 0
+        self.image = None
+        if len(self.results):
+            self.image = self.wolfie.visual_answer(query)
+            return self.results[0]["summary"]
+
+    def display_wolfie(self):
+        image = self.results[self.idx].get("img") or\
```

Comment:
Should this check if a gui is available before getting an image to display?

Reply:
i did this change locally already but didnt push yet